webrtc.html

```diff
@@ -9684,26 +9684,34 @@ <h4 id="sctp-transport-connected">Connected procedure</h4>
               <p>Set <a>[[\MaxChannels]]</a> to the minimum of the negotiated
               amount of incoming and outgoing SCTP streams.</p>
             </li>
-            <li>
-              <p><a>Fire an event</a> named <code><a data-lt="
-              RTCSctpTransport state change">statechange</a></code> at <var>
-              transport</var>.</p>
-            </li>
             <li>
               <p>For each of <var>connection</var>'s <code><a>RTCDataChannel</a></code>:</p>
               <ol>
                 <li>
                   <p>Let <var>channel</var> be the <code><a>RTCDataChannel</a></code> object.</p>
                 </li>
                 <li>
-                  <p>If the <var>channel</var>.<a>[[\DataChannelId]]</a> is greater or
-                  equal to <var>transport</var>.<a>[[\MaxChannels]]</a>,
+                  <p>If <var>channel</var>'s <a>[[\DataChannelId]]</a> slot is
+                    <code>null</code>, initialize <a>[[\DataChannelId]]</a>
+                    to the value generated by the
+                    underlying sctp data channel, according to
+                    [[!RTCWEB-DATA-PROTOCOL]].
+                  </p>
+                <li>
+                  <p>If <var>channel</var>'s <a>[[\DataChannelId]]</a> slot is greater or
+                    equal to <var>transport</var>'s <a>[[\MaxChannels]]</a> slot,
+                    or the previous step failed to assign an id,
                   <a data-lt="data-channel-failure"> close the <var>channel</var> due to a failure
                   </a>. Otherwise, <a data-lt="announce the rtcdatachannel as open">announce the
                   <var>channel</var> as open</a>.</p>
                 </li>
               </ol>
             </li>
+            <li>
+              <p><a>Fire an event</a> named <code><a data-lt="
+              RTCSctpTransport state change">statechange</a></code> at <var>
```

Comment:
The event fire order can now be misinterpreted easily. The individual `open` events fire after the `statechange` event because the `open` event is scheduled in yet another task. When we refactor that as suggested by @henbos, we need to ensure not to break that order: `statechange`, **then** for each data channel, `open`.

Reply:
Adding a note seems reasonable.